### PR TITLE
Allow complex database rules config

### LINKED ISSUE #2184: Database rules are not loaded when it uses `target`
### [REQUIRED] Environment info

**firebase-tools:** 8.2.0

**Platform:** Windows

### [REQUIRED] Test case

`firebase.json`:
```
{
  "database": [
    {
      "target": "default",
      "rules": "database.rules.json"
    }
  ],
  "emulators": {
    "database": {
      "port": 9000
    }
  }
}
```
`database.rules.json`:
```
{
  "rules": {
    ".write": false,
    ".read": false
  }
}
```

### [REQUIRED] Steps to reproduce

When using multiple shards, the documentation in https://firebase.google.com/docs/database/usage/sharding suggests adding an alternate form of database configuration that specifies targets. The emulator does not seem to recognize this format.

Run `firebase emulators:start --only database` with the files above. You will see the following warning:
```
!  database: Did not find a Realtime Database rules file specified in a firebase.json config file.
!  database: The emulator will default to allowing all reads and writes. Learn more about this option: https://firebase.google.com/docs/emulator-suite/install_and_configure#security_rules_configuration.
```

If you change the format back to `"database": {"rules": "database.rules.json"},`, the warning goes away and the security rules are loaded.

### [REQUIRED] Expected behavior

Database rules are not loaded.

### [REQUIRED] Actual behavior

Database rules are loaded.

### Other information

I'm happy enough using some other way to load security rules to multiple databases that doesn't involve targets. But it's unfortunate that at the moment I have to choose between being able to deploy to multiple databases or use emulators for testing.

### PR DESCRIPTION
**Pull request by samtstern**
### Description

Fixes #2184

Before this the emulator only allowed DB config in this format:

```json
 {
   "database" : {
    "rules": "...path..."
  }
}
```

Now we allow this:
```json
{
  "database": [
    {
      "target": "default",
      "rules": "database.rules.json"
    }
  ]
}
```

### Scenarios Tested

firebase.json
```json
{
  "database": [
    {
      "target": "default",
      "rules": "database.rules.json"
    }
  ]
}
```

.firebaserc:
```json
{
  "projects": {
    "default": "fir-dumpster"
  },
  "targets": {
    "fir-dumpster": {
      "database": {
        "default": [
          "fir-dumpster",
          "fir-dumpster-secondary"
        ]
      }
    }
  }
}
```

rules deploy:
```bash
$ firebase deploy --only database

=== Deploying to 'fir-dumpster'...

i  deploying database
i  database: checking rules syntax...
✔  database: rules syntax for database fir-dumpster is valid
✔  database: rules syntax for database fir-dumpster-secondary is valid
i  database: releasing rules...
✔  database: rules for database fir-dumpster-secondary released successfully
✔  database: rules for database fir-dumpster released successfully
```

emulators:
```
$ firebase emulators:start --only database
i  emulators: Starting emulators: database
✔  hub: emulator hub started at http://localhost:4400
i  database: database emulator logging to database-debug.log
✔  database: database emulator started at http://localhost:9000
i  database: For testing set FIREBASE_DATABASE_EMULATOR_HOST=localhost:9000
✔  emulators: All emulators started, it is now safe to connect.
i  database: Change detected, updating rules for fir-dumpster...
i  database: Change detected, updating rules for fir-dumpster-secondary...
✔  database: Rules updated.
✔  database: Rules updated.
^Ci  emulators: Shutting down emulators.
i  hub: Stopping emulator hub
i  database: Stopping database emulator
```

### CODE REVIEW

File: src/database/rulesConfig.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+interface DatabaseConfig {
+  rules: string;
+  target?: string;
+  instance?: string;
+}
+
+export function getRulesConfig(projectId: string, options: any): RulesInstanceConfig[] {
+  // TODO(samstern): Use the real config type here
```
Comment (samtstern) [inline src/database/rulesConfig.ts:16]: If I do `const config: Config = options.config` ts tries to tell me that there's no `.get(...)` function so I assume my import is wrong?

Comment (bkendall) [reply src/database/rulesConfig.ts:16]: Maybe do `options.config as Config`? Or, you can use a type guard: https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types

Comment (samtstern) [reply src/database/rulesConfig.ts:16]: I have tried about 50 different kinds of imports, declarations, and guards.  For some reason TS can see the properties of config from the constructor (`.data`, etc) but not any of the functions.  I declare defeat.

Comment (bkendall) [reply src/database/rulesConfig.ts:16]: 🤷 - sometimes the easiest way is to `any` it

**Pull request merged**